substrate/client/network/src/litep2p/shim/request_response/mod.rs

```diff
@@ -371,10 +373,33 @@ impl RequestResponseProtocol {
 
 		let status = match error {
 			RequestResponseError::NotConnected =>
-				Some((RequestFailure::NotConnected, "not-connected")),
-			RequestResponseError::Rejected => Some((RequestFailure::Refused, "rejected")),
+				Some((RequestFailure::NotConnected, "not-connected".to_string())),
+			RequestResponseError::Rejected(reason) => {
+				let reason = match reason {
+					RejectReason::ConnectionClosed => "connection-closed".to_string(),
```

Comment:
Also better to use a `Cow<'static, str>` here. (Less allocations)

Reply:
Have removed the allocations entirely, since we don't really need to expose each variant of `format!("substream-open-error: {:?}", substream_error)`